src/Pos/Delegation/Logic.hs

```diff
@@ -181,6 +182,19 @@ getProxyMempool = do
     toRollback <- catMaybes <$> mapM GS.getPSKByIssuer issuers
     pure (sks, toRollback)
 
+clearProxyMemPool
+    :: (MonadDB m, MonadDelegation m)
+    => m ()
+clearProxyMemPool = do
+    var <- askDelegationState
+    atomically $ modifyTVar' var resetData
+  where
+    resetData delegationWrap =
+        -- @volhovm is this correct or I must clear caches too?
+        delegationWrap {_dwProxySKPool = mempty,
+                        _dwThisEpochPosted = mempty
```

Comment:
Wrong.
You shouldn't clean anything except for these two. `dwProxySKPool` should be set to `mempty`, but you misunderstood the meaning of `dwThisEpochPosted`. It must not be set to `mempty` because it contains stakeholders that put their delegation certs to blockchain in main blocks of current epoch. Nothing to do with mempool.